### PR TITLE
Set global Typography default variant mappings to use <div>

### PR DESCRIPTION
**Pull request by 2metres**
**User-Facing Changes**
None

**Description**
Background: https://mui.com/material-ui/react-typography/#changing-the-semantic-element

The fundamental change here is:
```js 
    MuiTypography: {
      defaultProps: {
        variantMapping: {
          h1: "div",
          h2: "div",
          h3: "div",
          h4: "div",
          h5: "div",
          h6: "div",
          subtitle1: "div",
          subtitle2: "div",
          body1: "div",
          body2: "div",
          inherit: "div",
        },
      },
    },
```

### CODE REVIEW

**Comment by 2metres** [comment]
Reverting change because we do have instances where elements are being rendered inside the heading typographhy style

File: packages/studio-base/src/components/Sidebar/MemoryUseIndicator.tsx [hunk 1]
```diff
@@ -36,12 +36,10 @@ function MemoryUseIndicator(): JSX.Element {
...
     <Tooltip title={`Used (MB): ${usedMb} / ${limitMb}`}>
       <div className={classes.root}>
-        <Typography component="div" variant="caption" style={{ fontWeight: "bold" }}>
+        <Typography variant="caption" style={{ fontWeight: "bold" }}>
           mem
         </Typography>
-        <Typography component="div" variant="caption">
-          {usedPercent.toFixed(2)}%
-        </Typography>
+        <Typography variant="caption">{usedPercent.toFixed(2)}%</Typography>
```
Comment (jtbandes) [inline packages/studio-base/src/components/Sidebar/MemoryUseIndicator.tsx:42]: Need to add a space here

File: packages/studio-base/src/theme/muiComponents.ts [hunk 2]
```diff
@@ -535,5 +535,22 @@ export default function muiComponents(theme: Theme): Theme["components"] & MuiLa
...
+          h5: "div",
+          h6: "div",
+          subtitle1: "div",
+          subtitle2: "div",
+          body1: "div",
+          body2: "div",
+          inherit: "div",
+        },
+      },
+    },
```
Comment (jtbandes) [inline packages/studio-base/src/theme/muiComponents.ts:554]: Add a comment explaining why we do this

File: packages/studio-base/src/components/Sidebar/MemoryUseIndicator.tsx (hunk 1 above)
Comment (2metres) [reply packages/studio-base/src/components/Sidebar/MemoryUseIndicator.tsx:42]: if we leave them as div's it solves this issue. the default component for `variant="caption"` is a span

Comment (jtbandes) [reply packages/studio-base/src/components/Sidebar/MemoryUseIndicator.tsx:42]: Sure, that works

**Review by defunctzombie — APPROVED** [review]
Seems fine to me. I believe that <div> is the only true component.

**Comment by 2metres** [comment]
@defunctzombie this ☝️ is beautiful

**Pull request merged**